bin/post_peak_calling_processing.r

```diff
@@ -27,46 +28,24 @@ if (!require("ChIPpeakAnno")){
     library("ChIPpeakAnno")
 }
 
-if (!require("biomaRt")){
-    source("http://bioconductor.org/biocLite.R")
-    biocLite("biomaRt", suppressUpdates=TRUE)
-    library("biomaRt")
+# Process annotation file
+annotation<-read.table(Annotation,header=FALSE)
+annotation<-annotation[!duplicated(annotation),]
+genes<-unique(as.character(annotation$V5))
+anno<-matrix(nrow=length(genes),ncol=6)
+anno<-as.data.frame(anno)
+colnames(anno)<-c("chr","start","end","strand","gene","symbol")
+anno$gene<-genes
+
+for(i in 1:nrow(anno)){
+  anno$chr[i]<-unique(as.character(annotation[as.character(annotation$V5)==anno$gene[i],]$V1))
+  anno$start[i]<-min(as.numeric(as.character(annotation[as.character(annotation$V5)==anno$gene[i],]$V2)))
+  anno$end[i]<-max(as.numeric(as.character(annotation[as.character(annotation$V5)==anno$gene[i],]$V3)))
+  anno$strand[i]<-unique(as.character(annotation[as.character(annotation$V5)==anno$gene[i],]$V4))
+  anno$symbol[i]<-unique(as.character(annotation[as.character(annotation$V5)==anno$gene[i],]$V6))
 }
 
-# Load / install required annotation databases
-if(ref=="hs"){
-	if (!require("BSgenome.Hsapiens.UCSC.hg19")){
-        source("http://bioconductor.org/biocLite.R")
-        biocLite("BSgenome.Hsapiens.UCSC.hg19", suppressUpdates=TRUE)
-        library("BSgenome.Hsapiens.UCSC.hg19")
-    }
-    if (!require("org.Hs.eg.db")){
-        source("http://bioconductor.org/biocLite.R")
-        biocLite("org.Hs.eg.db", suppressUpdates=TRUE)
-        library("org.Hs.eg.db")
-    }
-	data(TSS.human.GRCh37)
-	annoData <- annoGR(TSS.human.GRCh37, feature="gene")
-	orgAnnData <-"org.Hs.eg.db"
-	mart=useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
-} else if(ref=="mm"){
-	if (!require("BSgenome.Mmusculus.UCSC.mm10")){
-        source("http://bioconductor.org/biocLite.R")
-        biocLite("BSgenome.Mmusculus.UCSC.mm10", suppressUpdates=TRUE)
-        library("BSgenome.Mmusculus.UCSC.mm10")
-    }
-    if (!require("org.Mm.eg.db")){
-        source("http://bioconductor.org/biocLite.R")
-        biocLite("org.Mm.eg.db", suppressUpdates=TRUE)
-        library("org.Mm.eg.db")
-    }
-	data(TSS.mouse.GRCm38)
-	annoData <- annoGR(TSS.mouse.GRCm38, feature="gene")
-	orgAnnData <-"org.Mm.eg.db"
-	mart=useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")
-} else{
-	stop("Genome not supported!", call.=FALSE)
-}
+annoData<-with(anno,GRanges(seqnames=chr,ranges=IRanges(start=start,end=end,names=gene),strand=strand,symbol=symbol))
```

Comment:
Kudos for writing all of the code to parse this from the GTF manually! However, it may be cleaner and potentially more robust to use an existing R package to do it.. For example:

```r
library('rtracklayer')
my_obj <- import(Annotation)
```

Here, `my_obj` should be a `GRanges` object (snippet found [here](https://davetang.org/muse/2017/08/04/read-gtf-file-r/) from a quick google. There seem to be other packages that can do it too).

Reply:
Hi, the GTF files on Uppmax are based on transcripts but not genes. Even though we read it with R it has to be manipulated.

Reply:
But thanks for the suggestions! It is indeed much better to import the gtf file in this way rather than with a separated step. I have changed the codes.

Reply:
Aha, that's what you're doing here. Ok fair enough :)